android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java

```diff
@@ -86,6 +86,7 @@ public class BravePrivacySettings extends PrivacySettings implements ConnectionE
     public static final String PREF_FINGERPRINTING_PROTECTION2 = "fingerprinting_protection2";
     private static final String PREF_CLOSE_TABS_ON_EXIT = "close_tabs_on_exit";
     private static final String PREF_SEND_P3A = "send_p3a_analytics";
+    private static final String PREF_SEND_WEB_DISCOVERY = "send_web_discovery";
```

Comment:
There is a mechanism to expose cpp prefs to java, but I don't recall what it is off-hand. @SergeyZhukovsky ?

Reply:
prefs from cpp to java are exposed via that section https://github.com/brave/brave-core/blob/master/browser/android/preferences/BUILD.gn#L37. You basically add files with prefs that you want to expose and the script does. The prefs could be found in an auto generated BrsavePref.java file that is made from a template https://github.com/brave/brave-core/blob/master/browser/preferences/android/java_templates/BravePref.java.tmpl
That's an example on how a pref would look like in Java
`public static final String BRAVE_CHAT_PURCHASE_TOKEN_ANDROID = "brave.ai_chat.purchase_token_android"`

Reply:
looks like we already do this:

https://github.com/brave/brave-core/blob/072fdfc3176b60fd71cedf9c05ed83a3420e7415/android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java#L334-L340

https://github.com/brave/brave-core/blob/072fdfc3176b60fd71cedf9c05ed83a3420e7415/browser/android/preferences/BUILD.gn#L37-L57